// @elevenlabs/elevenlabs-js/dist/serialization/resources/conversationalAi/resources/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.sipTrunk = exports.batchCalls = exports.secrets = exports.settings = exports.llmUsage = exports.agents = exports.twilio = exports.knowledgeBase = exports.dashboard = exports.mcpServers = exports.phoneNumbers = exports.tests = exports.conversations = void 0;
exports.conversations = __importStar(require("./conversations"));
__exportStar(require("./conversations/types"), exports);
exports.tests = __importStar(require("./tests"));
__exportStar(require("./tests/types"), exports);
exports.phoneNumbers = __importStar(require("./phoneNumbers"));
__exportStar(require("./phoneNumbers/types"), exports);
exports.mcpServers = __importStar(require("./mcpServers"));
__exportStar(require("./mcpServers/types"), exports);
exports.dashboard = __importStar(require("./dashboard"));
exports.knowledgeBase = __importStar(require("./knowledgeBase"));
exports.twilio = __importStar(require("./twilio"));
__exportStar(require("./twilio/client/requests"), exports);
exports.agents = __importStar(require("./agents"));
__exportStar(require("./agents/client/requests"), exports);
__exportStar(require("./tests/client/requests"), exports);
__exportStar(require("./phoneNumbers/client/requests"), exports);
exports.llmUsage = __importStar(require("./llmUsage"));
__exportStar(require("./llmUsage/client/requests"), exports);
exports.settings = __importStar(require("./settings"));
__exportStar(require("./settings/client/requests"), exports);
exports.secrets = __importStar(require("./secrets"));
__exportStar(require("./secrets/client/requests"), exports);
exports.batchCalls = __importStar(require("./batchCalls"));
__exportStar(require("./batchCalls/client/requests"), exports);
exports.sipTrunk = __importStar(require("./sipTrunk"));
__exportStar(require("./sipTrunk/client/requests"), exports);
__exportStar(require("./mcpServers/client/requests"), exports);
