// @elevenlabs/elevenlabs-js/dist/serialization/types/AstLessThanOrEqualsOperatorNodeInputRight.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AstLessThanOrEqualsOperatorNodeInputRight = void 0;
const serializers = __importStar(require("../index"));
const core = __importStar(require("../../core"));
const AstBooleanNodeInput_1 = require("./AstBooleanNodeInput");
const AstDynamicVariableNodeInput_1 = require("./AstDynamicVariableNodeInput");
const AstllmNodeInput_1 = require("./AstllmNodeInput");
const AstNumberNodeInput_1 = require("./AstNumberNodeInput");
const AstStringNodeInput_1 = require("./AstStringNodeInput");
exports.AstLessThanOrEqualsOperatorNodeInputRight = core.serialization
    .union("type", {
    and_operator: core.serialization.lazyObject(() => serializers.AstAndOperatorNodeInput),
    boolean_literal: AstBooleanNodeInput_1.AstBooleanNodeInput,
    dynamic_variable: AstDynamicVariableNodeInput_1.AstDynamicVariableNodeInput,
    eq_operator: core.serialization.lazyObject(() => serializers.AstEqualsOperatorNodeInput),
    gt_operator: core.serialization.lazyObject(() => serializers.AstGreaterThanOperatorNodeInput),
    gte_operator: core.serialization.lazyObject(() => serializers.AstGreaterThanOrEqualsOperatorNodeInput),
    llm: AstllmNodeInput_1.AstllmNodeInput,
    lt_operator: core.serialization.lazyObject(() => serializers.AstLessThanOperatorNodeInput),
    lte_operator: core.serialization.lazyObject(() => serializers.AstLessThanOrEqualsOperatorNodeInput),
    neq_operator: core.serialization.lazyObject(() => serializers.AstNotEqualsOperatorNodeInput),
    number_literal: AstNumberNodeInput_1.AstNumberNodeInput,
    or_operator: core.serialization.lazyObject(() => serializers.AstOrOperatorNodeInput),
    string_literal: AstStringNodeInput_1.AstStringNodeInput,
})
    .transform({
    transform: (value) => value,
    untransform: (value) => value,
});
