// @elevenlabs/elevenlabs-js/dist/serialization/types/CaptionStyleModel.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaptionStyleModel = void 0;
const core = __importStar(require("../../core"));
const CaptionStyleTemplateModel_1 = require("./CaptionStyleTemplateModel");
const CaptionStyleModelTextAlign_1 = require("./CaptionStyleModelTextAlign");
const CaptionStyleModelTextStyle_1 = require("./CaptionStyleModelTextStyle");
const CaptionStyleModelTextWeight_1 = require("./CaptionStyleModelTextWeight");
const CaptionStyleSectionAnimationModel_1 = require("./CaptionStyleSectionAnimationModel");
const CaptionStyleWordAnimationModel_1 = require("./CaptionStyleWordAnimationModel");
const CaptionStyleCharacterAnimationModel_1 = require("./CaptionStyleCharacterAnimationModel");
const CaptionStyleHorizontalPlacementModel_1 = require("./CaptionStyleHorizontalPlacementModel");
const CaptionStyleVerticalPlacementModel_1 = require("./CaptionStyleVerticalPlacementModel");
exports.CaptionStyleModel = core.serialization.object({
    template: CaptionStyleTemplateModel_1.CaptionStyleTemplateModel.optional(),
    textFont: core.serialization.property("text_font", core.serialization.string().optional()),
    textScale: core.serialization.property("text_scale", core.serialization.number().optional()),
    textColor: core.serialization.property("text_color", core.serialization.string().optional()),
    textAlign: core.serialization.property("text_align", CaptionStyleModelTextAlign_1.CaptionStyleModelTextAlign.optional()),
    textStyle: core.serialization.property("text_style", CaptionStyleModelTextStyle_1.CaptionStyleModelTextStyle.optional()),
    textWeight: core.serialization.property("text_weight", CaptionStyleModelTextWeight_1.CaptionStyleModelTextWeight.optional()),
    backgroundEnabled: core.serialization.property("background_enabled", core.serialization.boolean().optional()),
    backgroundColor: core.serialization.property("background_color", core.serialization.string().optional()),
    backgroundOpacity: core.serialization.property("background_opacity", core.serialization.number().optional()),
    wordHighlightsEnabled: core.serialization.property("word_highlights_enabled", core.serialization.boolean().optional()),
    wordHighlightsColor: core.serialization.property("word_highlights_color", core.serialization.string().optional()),
    wordHighlightsBackgroundColor: core.serialization.property("word_highlights_background_color", core.serialization.string().optional()),
    wordHighlightsOpacity: core.serialization.property("word_highlights_opacity", core.serialization.number().optional()),
    sectionAnimation: core.serialization.property("section_animation", CaptionStyleSectionAnimationModel_1.CaptionStyleSectionAnimationModel.optional()),
    wordAnimation: core.serialization.property("word_animation", CaptionStyleWordAnimationModel_1.CaptionStyleWordAnimationModel.optional()),
    characterAnimation: core.serialization.property("character_animation", CaptionStyleCharacterAnimationModel_1.CaptionStyleCharacterAnimationModel.optional()),
    widthPct: core.serialization.property("width_pct", core.serialization.number().optional()),
    horizontalPlacement: core.serialization.property("horizontal_placement", CaptionStyleHorizontalPlacementModel_1.CaptionStyleHorizontalPlacementModel.optional()),
    verticalPlacement: core.serialization.property("vertical_placement", CaptionStyleVerticalPlacementModel_1.CaptionStyleVerticalPlacementModel.optional()),
    autoBreakEnabled: core.serialization.property("auto_break_enabled", core.serialization.boolean().optional()),
    maxLinesPerSection: core.serialization.property("max_lines_per_section", core.serialization.number().optional()),
    maxWordsPerLine: core.serialization.property("max_words_per_line", core.serialization.number().optional()),
});
