// @elevenlabs/elevenlabs-js/dist/serialization/resources/conversationalAi/resources/tests/client/requests/CreateUnitTestRequest.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateUnitTestRequest = void 0;
const core = __importStar(require("../../../../../../../core"));
const ConversationHistoryTranscriptCommonModelInput_1 = require("../../../../../../types/ConversationHistoryTranscriptCommonModelInput");
const AgentSuccessfulResponseExample_1 = require("../../../../../../types/AgentSuccessfulResponseExample");
const AgentFailureResponseExample_1 = require("../../../../../../types/AgentFailureResponseExample");
const UnitTestToolCallEvaluationModelInput_1 = require("../../../../../../types/UnitTestToolCallEvaluationModelInput");
const CreateUnitTestRequestDynamicVariablesValue_1 = require("../../types/CreateUnitTestRequestDynamicVariablesValue");
const UnitTestCommonModelType_1 = require("../../../../../../types/UnitTestCommonModelType");
const TestFromConversationMetadataInput_1 = require("../../../../../../types/TestFromConversationMetadataInput");
exports.CreateUnitTestRequest = core.serialization.object({
    chatHistory: core.serialization.property("chat_history", core.serialization.list(ConversationHistoryTranscriptCommonModelInput_1.ConversationHistoryTranscriptCommonModelInput)),
    successCondition: core.serialization.property("success_condition", core.serialization.string()),
    successExamples: core.serialization.property("success_examples", core.serialization.list(AgentSuccessfulResponseExample_1.AgentSuccessfulResponseExample)),
    failureExamples: core.serialization.property("failure_examples", core.serialization.list(AgentFailureResponseExample_1.AgentFailureResponseExample)),
    toolCallParameters: core.serialization.property("tool_call_parameters", UnitTestToolCallEvaluationModelInput_1.UnitTestToolCallEvaluationModelInput.optional()),
    dynamicVariables: core.serialization.property("dynamic_variables", core.serialization
        .record(core.serialization.string(), CreateUnitTestRequestDynamicVariablesValue_1.CreateUnitTestRequestDynamicVariablesValue.optional())
        .optional()),
    type: UnitTestCommonModelType_1.UnitTestCommonModelType.optional(),
    fromConversationMetadata: core.serialization.property("from_conversation_metadata", TestFromConversationMetadataInput_1.TestFromConversationMetadataInput.optional()),
    name: core.serialization.string(),
});
