// @elevenlabs/elevenlabs-js/serialization/types/AgentWorkflowResponseModelNodesValue.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentWorkflowResponseModelNodesValue = void 0;
const core = __importStar(require("../../core"));
const WorkflowEndNodeModelOutput_1 = require("./WorkflowEndNodeModelOutput");
const WorkflowOverrideAgentNodeModelOutput_1 = require("./WorkflowOverrideAgentNodeModelOutput");
const WorkflowPhoneNumberNodeModelOutput_1 = require("./WorkflowPhoneNumberNodeModelOutput");
const WorkflowStandaloneAgentNodeModelOutput_1 = require("./WorkflowStandaloneAgentNodeModelOutput");
const WorkflowStartNodeModelOutput_1 = require("./WorkflowStartNodeModelOutput");
const WorkflowToolNodeModelOutput_1 = require("./WorkflowToolNodeModelOutput");
exports.AgentWorkflowResponseModelNodesValue = core.serialization
    .union("type", {
    end: WorkflowEndNodeModelOutput_1.WorkflowEndNodeModelOutput,
    override_agent: WorkflowOverrideAgentNodeModelOutput_1.WorkflowOverrideAgentNodeModelOutput,
    phone_number: WorkflowPhoneNumberNodeModelOutput_1.WorkflowPhoneNumberNodeModelOutput,
    standalone_agent: WorkflowStandaloneAgentNodeModelOutput_1.WorkflowStandaloneAgentNodeModelOutput,
    start: WorkflowStartNodeModelOutput_1.WorkflowStartNodeModelOutput,
    tool: WorkflowToolNodeModelOutput_1.WorkflowToolNodeModelOutput,
})
    .transform({
    transform: (value) => value,
    untransform: (value) => value,
});
