// @elevenlabs/elevenlabs-js/serialization/types/AstLessThanOperatorNodeOutputLeft.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AstLessThanOperatorNodeOutputLeft = void 0;
const serializers = __importStar(require("../index"));
const core = __importStar(require("../../core"));
const AstBooleanNodeOutput_1 = require("./AstBooleanNodeOutput");
const AstDynamicVariableNodeOutput_1 = require("./AstDynamicVariableNodeOutput");
const AstllmNodeOutput_1 = require("./AstllmNodeOutput");
const AstNumberNodeOutput_1 = require("./AstNumberNodeOutput");
const AstStringNodeOutput_1 = require("./AstStringNodeOutput");
exports.AstLessThanOperatorNodeOutputLeft = core.serialization
    .union("type", {
    and_operator: core.serialization.lazyObject(() => serializers.AstAndOperatorNodeOutput),
    boolean_literal: AstBooleanNodeOutput_1.AstBooleanNodeOutput,
    dynamic_variable: AstDynamicVariableNodeOutput_1.AstDynamicVariableNodeOutput,
    eq_operator: core.serialization.lazyObject(() => serializers.AstEqualsOperatorNodeOutput),
    gt_operator: core.serialization.lazyObject(() => serializers.AstGreaterThanOperatorNodeOutput),
    gte_operator: core.serialization.lazyObject(() => serializers.AstGreaterThanOrEqualsOperatorNodeOutput),
    llm: AstllmNodeOutput_1.AstllmNodeOutput,
    lt_operator: core.serialization.lazyObject(() => serializers.AstLessThanOperatorNodeOutput),
    lte_operator: core.serialization.lazyObject(() => serializers.AstLessThanOrEqualsOperatorNodeOutput),
    neq_operator: core.serialization.lazyObject(() => serializers.AstNotEqualsOperatorNodeOutput),
    number_literal: AstNumberNodeOutput_1.AstNumberNodeOutput,
    or_operator: core.serialization.lazyObject(() => serializers.AstOrOperatorNodeOutput),
    string_literal: AstStringNodeOutput_1.AstStringNodeOutput,
})
    .transform({
    transform: (value) => value,
    untransform: (value) => value,
});
