// @elevenlabs/elevenlabs-js/serialization/types/ConversationHistoryTranscriptSystemToolResultCommonModelResult.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversationHistoryTranscriptSystemToolResultCommonModelResult = void 0;
const core = __importStar(require("../../core"));
const EndCallToolResultModel_1 = require("./EndCallToolResultModel");
const LanguageDetectionToolResultModel_1 = require("./LanguageDetectionToolResultModel");
const PlayDtmfResultErrorModel_1 = require("./PlayDtmfResultErrorModel");
const PlayDtmfResultSuccessModel_1 = require("./PlayDtmfResultSuccessModel");
const SkipTurnToolResponseModel_1 = require("./SkipTurnToolResponseModel");
const TestToolResultModel_1 = require("./TestToolResultModel");
const TransferToAgentToolResultErrorModel_1 = require("./TransferToAgentToolResultErrorModel");
const TransferToAgentToolResultSuccessModel_1 = require("./TransferToAgentToolResultSuccessModel");
const TransferToNumberResultErrorModel_1 = require("./TransferToNumberResultErrorModel");
const TransferToNumberResultSipSuccessModel_1 = require("./TransferToNumberResultSipSuccessModel");
const TransferToNumberResultTwilioSuccessModel_1 = require("./TransferToNumberResultTwilioSuccessModel");
const VoiceMailDetectionResultSuccessModel_1 = require("./VoiceMailDetectionResultSuccessModel");
exports.ConversationHistoryTranscriptSystemToolResultCommonModelResult = core.serialization
    .union(core.serialization.discriminant("resultType", "result_type"), {
    end_call_success: EndCallToolResultModel_1.EndCallToolResultModel,
    language_detection_success: LanguageDetectionToolResultModel_1.LanguageDetectionToolResultModel,
    play_dtmf_error: PlayDtmfResultErrorModel_1.PlayDtmfResultErrorModel,
    play_dtmf_success: PlayDtmfResultSuccessModel_1.PlayDtmfResultSuccessModel,
    skip_turn_success: SkipTurnToolResponseModel_1.SkipTurnToolResponseModel,
    testing_tool_result: TestToolResultModel_1.TestToolResultModel,
    transfer_to_agent_error: TransferToAgentToolResultErrorModel_1.TransferToAgentToolResultErrorModel,
    transfer_to_agent_success: TransferToAgentToolResultSuccessModel_1.TransferToAgentToolResultSuccessModel,
    transfer_to_number_error: TransferToNumberResultErrorModel_1.TransferToNumberResultErrorModel,
    transfer_to_number_sip_success: TransferToNumberResultSipSuccessModel_1.TransferToNumberResultSipSuccessModel,
    transfer_to_number_twilio_success: TransferToNumberResultTwilioSuccessModel_1.TransferToNumberResultTwilioSuccessModel,
    voicemail_detection_success: VoiceMailDetectionResultSuccessModel_1.VoiceMailDetectionResultSuccessModel,
})
    .transform({
    transform: (value) => value,
    untransform: (value) => value,
});
