// @elevenlabs/elevenlabs-js/serialization/types/ModerationConfig.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModerationConfig = void 0;
const core = __importStar(require("../../core"));
const ThresholdGuardrail_1 = require("./ThresholdGuardrail");
exports.ModerationConfig = core.serialization.object({
    sexual: ThresholdGuardrail_1.ThresholdGuardrail.optional(),
    violence: ThresholdGuardrail_1.ThresholdGuardrail.optional(),
    violenceGraphic: core.serialization.property("violence_graphic", ThresholdGuardrail_1.ThresholdGuardrail.optional()),
    harassment: ThresholdGuardrail_1.ThresholdGuardrail.optional(),
    harassmentThreatening: core.serialization.property("harassment_threatening", ThresholdGuardrail_1.ThresholdGuardrail.optional()),
    hate: ThresholdGuardrail_1.ThresholdGuardrail.optional(),
    hateThreatening: core.serialization.property("hate_threatening", ThresholdGuardrail_1.ThresholdGuardrail.optional()),
    selfHarmInstructions: core.serialization.property("self_harm_instructions", ThresholdGuardrail_1.ThresholdGuardrail.optional()),
    selfHarm: core.serialization.property("self_harm", ThresholdGuardrail_1.ThresholdGuardrail.optional()),
    selfHarmIntent: core.serialization.property("self_harm_intent", ThresholdGuardrail_1.ThresholdGuardrail.optional()),
    sexualMinors: core.serialization.property("sexual_minors", ThresholdGuardrail_1.ThresholdGuardrail.optional()),
});
