// @elevenlabs/elevenlabs-js/serialization/types/SystemToolConfigInput.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.SystemToolConfigInput = void 0;
const core = __importStar(require("../../core"));
const DynamicVariableAssignment_1 = require("./DynamicVariableAssignment");
const ToolCallSoundType_1 = require("./ToolCallSoundType");
const ToolCallSoundBehavior_1 = require("./ToolCallSoundBehavior");
const SystemToolConfigInputParams_1 = require("./SystemToolConfigInputParams");
exports.SystemToolConfigInput = core.serialization.object({
    name: core.serialization.string(),
    description: core.serialization.string().optional(),
    responseTimeoutSecs: core.serialization.property("response_timeout_secs", core.serialization.number().optional()),
    disableInterruptions: core.serialization.property("disable_interruptions", core.serialization.boolean().optional()),
    forcePreToolSpeech: core.serialization.property("force_pre_tool_speech", core.serialization.boolean().optional()),
    assignments: core.serialization.list(DynamicVariableAssignment_1.DynamicVariableAssignment).optional(),
    toolCallSound: core.serialization.property("tool_call_sound", ToolCallSoundType_1.ToolCallSoundType.optional()),
    toolCallSoundBehavior: core.serialization.property("tool_call_sound_behavior", ToolCallSoundBehavior_1.ToolCallSoundBehavior.optional()),
    params: SystemToolConfigInputParams_1.SystemToolConfigInputParams,
});
